mantidproject/mantid
# Fix LOQ Batch reduction issues

Fixes a set of batch mode related issues found by Steve King.

The fix here adresses:
* `FileFinder` being sensitive to file extensions
* Wrong file suffix for output workspace 
* Missing geometry options for `SaveCanSAS1D`

The fixes are being validated by Steve King -- Steve King wrote:
> So I hereby validate the three issues in the ticket… but I have discovered a new, or perhaps it is unsolved, issue that explains the mysterious Q-range…

The subsequent issue is addressed here: #18926

### To test:

#### Test LOQ

Steve King's LOQ instrument was having the issues so, let's test this first.
The test data can be found here:  `\\olympic\Babylon5\Scratch\Anton\LOQ_Batch_BUG_2`
Make sure it is in your Mantid path

1. Open the ISIS SANS GUI
2. Set the instrument to LOQ
3. Set the user file to USER_LOQ_165B_M3_XinXpress_4mm_Changer_MAIN.txt
4. In the save options enable `CanSAS (1D)`
5. Select the Batch mode button
6. Click browse and select the csv file: reduce-nimrod-check-samples-main.csv 
   * Confirm that the entries in `user_file` column don't have a `.txt` extension
7. Press `1DReduce`
   * Confirm that there are no errors
   * Confirm that the output of the reduction, ie workspaces with the name `Sample1_600C_*` contain the suffix `main`. You will see it twice here since the output name was specified as `Sample1_600C_main` which then gets a suffix `_main`. The important bit is that there is a suffix `_main`
8. Load the batch file `reduce-nimrod-check-samples-hab.csv `
     * Confirm that the entries in `user_file` column don't have a `.txt` extension
9. Press `1DReduce`
   * Confirm that there are no errors
   * Confirm that the output of the reduction, ie workspaces with the name `Sample1_600C_hab*` contain the suffix `_hab`. You will see it twice here since the output name was specified as `Sample1_600C_hab` which then gets a suffix `_hab`. The important bit is that there is a suffix `_hab`
10. Create a plot for `Sample1_600C_main_main` and  `Sample1_600C_hab_hab`.
  * Confirm the q-range for  `Sample1_600C_main_main` is `.007` to `0.255` 
  * Confirm the q-range for  `Sample1_600C_hab_hab` is `.1125 ` to `1.4`
11. The results were saved out in the `CanSAS1D` format. Open the file `Sample1_600C_hab_hab.xml` in a text editor.
  * Confirm that there is an entry 
```xml	
<SAScollimation>
  <aperture name="Disc">
    <size>
      <x unit="mm">4.000000</x>
      <y unit="mm">4.000000</y>
    </size>
   </aperture>
</SAScollimation>
```

  * Confirm that there is an entry: 
```xml		
<SASsample>
  <ID>Xin_Sample1_1200_2h_BQ_600C_2h_BQ_4mm_SANS</ID>
  <thickness unit="mm">1.550000</thickness>
</SASsample>
```

#### Test SANS2D

Please find the releveant test data here:  \\olympic\Babylon5\Scratch\Anton\SANS2DTUBES_Batch_BUG
Please make sure that the data is in your path.

1. Open the ISIS SANS GUI
2. Set the instrument to SANS2DTUBES
3. Set the user file to: USER_SANS2D_154E_2p4_4m_M3_Xpress_8mm_SampleChanger.txt
5. Select the Batch mode button
6. Click browse and select the csv file: batchTest.csv 
   * Confirm that the entries in `user_file` column don't have a `.txt` extension
7. Press `1DReduce`
   * Confirm that there are no errors
   * Confirm that the output of the reduction with the name `test_file_rear` exists. The important thing here is the suffix `_rear`.
8. Load the batch file `batchTestFront.csv `
     * Confirm that the entries in `user_file` column don't have a `.txt` extension
9. Press `1DReduce`
   * Confirm that there are no errors
   * Confirm that the output of the reduction with the name `test_file_front` exists. The important thing here is the suffix `_front`.
10. The results were saved out in the `CanSAS1D` format. Open the file `test_file_rear.xml` in a text editor.
  * Confirm that there is an entry 
```xml	
<SAScollimation>
  <aperture name="Disc">
    <size>
      <x unit="mm">8.000000</x>
      <y unit="mm">8.000000</y>
    </size>
   </aperture>
</SAScollimation>
```

  * Confirm that there is an entry: 
```xml		
<SASsample>
  <ID>XB1690177 1 Lipo _SANS</ID>
  <thickness unit="mm">2.000000</thickness>
</SASsample>
```



Fixes #18877 

### Release notes

Please see [here](https://github.com/mantidproject/mantid/pull/18888/commits/5441426e551fa2799e7846153de7811759a37cc2)


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
